OpenEdX/Router.swift

```diff
@@ -854,4 +866,114 @@ extension Router {
         navigationController.setViewControllers(viewControllers, animated: true)
     }
 }
+
+// MARK: Native alerts
+extension Router {
+    public func presentNativeAlert(title: String?, message: String?, actions: [UIAlertAction]) {
+        guard let topController = UIApplication.topViewController() else { return }
+        
+        let alertController = UIAlertController().showAlert(
+            withTitle: title,
+            message: message,
+            onViewController: topController) { _, _, _ in }
+        for action in actions {
+            alertController.addAction(action)
+        }
+    }
+}
+
+// MARK: Payments
+extension Router {
+    @MainActor
+    public func showUpgradeInfo(
+        productName: String,
+        message: String,
+        sku: String,
+        courseID: String,
+        screen: CourseUpgradeScreen,
+        pacing: String
+    ) async {
+        await withCheckedContinuation { continuation in
+            let view = UpgradeInfoSheetView(
+                viewModel: Container.shared.resolve(
+                    UpgradeInfoViewModel.self,
+                    arguments: productName, message, sku, courseID, screen, pacing
+                )!
+            )
+            let controller = UIHostingController(rootView: view)
+            if let sheet = controller.sheetPresentationController {
+                sheet.detents = [.large()]
+                sheet.prefersEdgeAttachedInCompactHeight = true
+                sheet.widthFollowsPreferredContentSizeWhenEdgeAttached = true
+                sheet.prefersGrabberVisible = true
+            }
+            navigationController.present(controller, animated: true) {
+                continuation.resume()
+            }
+        }
+    }
+    
+    @MainActor
+    public func hideUpgradeInfo(animated: Bool) async {
+        await withCheckedContinuation { continuation in
+            if let controller = navigationController.presentedViewController as? UIHostingController<UpgradeInfoSheetView> {
```

Comment:
Line Length Violation: Line should be 120 characters or less; currently it has 124 characters (line_length)